### PR TITLE
Highlight selected data picker expression.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
It's helpful to be able to see what expression is currently being used in the data picker.

**Fix:**
The entire hierarchy of the expression (should any apply) is italicised in the data picker if it matches.
image

For an expression we produce a simple path representation that is matched against the same path representation produced for the variables listed in the data picker.

**Commit Details:**
- `useDataPickerButton` takes two more properties so that `useComponentPropsInspectorInfo` can be called within it.
- Added `valuePath` field to `PrimitiveOption`, `ArrayOption`, `ObjectOption` and `JSXOption`.
- `DataPickerPopup` now takes an additional `propExpressionPath` property.
- `ValueRow` takes a `currentPropExpressionPath` to represent what is currently being used for a property if an appropriate expression is being used.
- Span in `ValueRow` gets italicised if the current expression matches.
- Slightly rejigged the fields of the `VariableInfo` types to make them a little clearer and get the appropriate values needed now.
- Added `isPrefixOf` utility function.
- Added `modifiableAttributeToValuePath` utility function.

### CODE REVIEW

File: editor/src/core/shared/element-template.ts [hunk 1]
```diff
@@ -961,6 +960,32 @@ export function isRegularJSXAttribute(attribute: ModifiableAttribute): attribute
   )
 }
 
+export function modifiableAttributeToValuePath(
```
Comment (balazsbajorics) [inline editor/src/core/shared/element-template.ts:963]: could you add some test coverage for this? I _think_ this function will be key for the data tracing features next week

Comment (seanparsons) [reply editor/src/core/shared/element-template.ts:963]: Yup.

File: editor/src/components/inspector/sections/component-section/data-picker-popup.tsx [hunk 2]
```diff
@@ -272,7 +294,7 @@ function ValueRow({ variableOption, idx, onTweakProperty }: ValueRowProps) {
                   opacity: variableOption.variableInfo.matches ? 1 : 0.5,
                 }}
               >
-                {variableOption.variableInfo.displayName}
```
Comment (balazsbajorics) [inline editor/src/components/inspector/sections/component-section/data-picker-popup.tsx:275]: is this a related change?

**Review by bkrmendy — APPROVED** [review]

**Review by balazsbajorics — APPROVED** [review]
